citycoin.clar

```diff
@@ -624,6 +624,7 @@
 )
 
 ;;;;;;;;;;;;;;;;;;;;; SIP 010 ;;;;;;;;;;;;;;;;;;;;;;
+(impl-trait 'SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE.sip-10-ft-standard.ft-trait)
```

Comment:
This should be at the top of the contract based on [the function reference](https://docs.stacks.co/references/language-functions#impl-trait)

> Like other kinds of definition statements, impl-trait may only be used at the top level of a smart contract definition...

Also, wouldn't the functions below the declaration need to be removed, since they are already covered by the SIP-010 contract?

Reply:
`impl-trait` must be used at the top level of **smart contract definition** not **smart contract**, which means we can put wherever we want as long as it is not inside other functions (it is not nested).
As Clarity contracts have 100+ lines (this one have 655) I thought it will be a good idea to keep `impl-trait` as close as possible to its implementation, so that you don't have to scroll to the top, to see address of you are implementing.
It's just a styling convention so if we decide to put all `impl-trait` at the top of the contract file - I will comply to such rule.

> Also, wouldn't the functions below the declaration need to be removed, since they are already covered by the SIP-010 contract?
No, they must stay there. 
Up to this point this contract was compliant to SIP-010 implicitly - which means that it was developers duty to make sure all functions listed in SIP-010 are present and defined correctly.
Adding `(impl-trait 'SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE.sip-10-ft-standard.ft-trait)` made this implementation explicit which means that if we remove any of those functions and try to deploy contract, Clarity interpreter will tell us that contract is invalid as one of the functions defined in trait is missing.

Reply:
got it - thank you for the clarification! In that case it makes sense to keep it where you have it.